src/test/java/com/puppycrawl/tools/checkstyle/PackageObjectFactoryTest.java

```diff
@@ -491,4 +528,8 @@ protected void processFiltered(File file, FileText fileText) {
 
     }
 
+    private static class MockClass {
+        MockClass() {
```

Comment:
Do we need explicit default constructor?

Reply:
If the class has an access modifier `private` then the default constructor by default also has a `private` access modifier, reflection can find that constructor (we won't get `NoSuchMethodException`) but we will get `IllegalAccessException`. So if we use private then the default constructor either needs to be `public` or with the default access modifier.

Reply:
@nick-mancuso Semaphore is complaining, `setAccessable(true)` won't work as I would have to change the code in `PackageObjectFactory` to do so.

Reply:
https://github.com/checkstyle/checkstyle/blob/399f1b48b79731d65e824eb927971ef6174c3463/src/main/java/com/puppycrawl/tools/checkstyle/utils/ModuleReflectionUtil.java#L96

Reply:
Let’s skip this, if constructor is not redundant, I am good.